### PR TITLE
FRONT-2483: Progress pattern.

### LINKED ISSUE #29: Add Progress-bar pattern with BCL component

### PR DESCRIPTION
**Pull request by escuriola**

### CODE REVIEW

**Review by drishu — CHANGES_REQUESTED** [review]
For some reason the progress isn't indicated, I have to manually refresh, please look into it.

File: templates/patterns/progress/pattern-progress.html.twig [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{#
+/**
+ * @file
+ * Progress.
+ */
+#}
+{% include '@oe-bcl/bcl-progress/progress.html.twig' with {
+  'progress': progress,
+  'striped': striped,
```
Comment (drishu) [inline templates/patterns/progress/pattern-progress.html.twig:12]: min, max, message are missing

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        warning: "warning"
+        info: "info"
+        light: "light"
+        dark: "dark"
+      preview: "primary"
+    striped:
+      type: "boolean"
+      label: "Striped style of state"
+      preview: true
+    animated:
```
Comment (drishu) [inline templates/patterns/progress/progress.ui_patterns.yml:22]: min, max are missing

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  fields:
+    label:
+      type: "text"
+      label: "Label in the bar"
+      preview: "This is the label of progress bar"
+    progress:
+      type: "text"
+      label: "Percentage of progress in the bar"
+      preview: "50"
+    bar_label:
```
Comment (drishu) [inline templates/patterns/progress/progress.ui_patterns.yml:35]: message is missing

File: tests/fixtures/markup_rendering_pattern.yml [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+    count:
+      'div[data-drupal-selector="progress"]': 1
+      'div.progress-bar': 1
+      'div.progress': 1
+      'div[role="progressbar"]': 1
+      'div[style="width: 60%"]': 1
+      'div[aria-valuenow="60"]': 1
+      'div[aria-valuemin="0"]': 1
+      'div[aria-valuemax="100"]': 1
+    contains:
```
Comment (drishu) [inline tests/fixtures/markup_rendering_pattern.yml:23]: check for .progress-bar-message contains what was set in message once its avail

File: tests/fixtures/markup_rendering_pattern.yml [hunk 5]
```diff
@@ -1,3 +1,24 @@
+progress:
+  render:
+    '#type': pattern
+    '#id': progress
+    '#fields':
+      progress: 60
+      label: "Progress label"
+      bar_label: "Bar label"
```
Comment (drishu) [inline tests/fixtures/markup_rendering_pattern.yml:8]: add message, min, max once avail

File: tests/fixtures/markup_rendering_pattern.yml [hunk 6]
```diff
@@ -1,3 +1,24 @@
...
+  assertions:
+    count:
+      'div[data-drupal-selector="progress"]': 1
+      'div.progress-bar': 1
+      'div.progress': 1
+      'div[role="progressbar"]': 1
```
Comment (drishu) [inline tests/fixtures/markup_rendering_pattern.yml:17]: check for min & max once avail

**Review by drishu — CHANGES_REQUESTED** [review]

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      type: "text"
+      label: "Text indicating the percentage of progress in the bar"
+      preview: "50%"
+    message:
+      type: "text"
+      label: "Optional description text below the progress bar"
+    min:
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml:42]: Please check, min should be moved to settings part.

File: templates/patterns/progress/progress.ui_patterns.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      type: "text"
+      label: "Minimum state of prorgress bar"
+      preview: "0"
+    max:
```
Comment (abel-santos-corral) [inline templates/patterns/progress/progress.ui_patterns.yml:46]: Same for max, should be placed at settings.

**Review by abel-santos-corral — COMMENTED** [review]
Please check the place where min and max are now, move them to settings.

File: templates/patterns/progress/progress.ui_patterns.yml (hunk 7 above)
Comment (escuriola) [reply templates/patterns/progress/progress.ui_patterns.yml:42]: Min and Max are fields due to free text is not allowed on the settings section and at alexis branch this were also fields, not settings.

File: templates/patterns/progress/progress.ui_patterns.yml (hunk 8 above)
Comment (escuriola) [reply templates/patterns/progress/progress.ui_patterns.yml:46]: Same as above

**Review by abel-santos-corral — DISMISSED** [review]
Reviewed. Passing to QA.

**Review by drishu — DISMISSED** [review]

**Review by drishu — APPROVED** [review]

**Review by abel-santos-corral — APPROVED** [review]

**Pull request merged**